src/main/java/uk/gov/pay/products/service/PaymentUpdater.java

```diff
@@ -0,0 +1,33 @@
+package uk.gov.pay.products.service;
+
+import com.google.inject.persist.Transactional;
+import org.slf4j.*;
+import uk.gov.pay.products.persistence.dao.PaymentDao;
+import uk.gov.pay.products.persistence.entity.PaymentEntity;
+
+import javax.inject.Inject;
+import java.util.Optional;
+
+import static java.lang.String.format;
+
+public class PaymentUpdater {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PaymentUpdater.class);
+    
+    public static final String REDACTED_REFERENCE_NUMBER = "****************";
```

Comment:
I counted: 16 asterisks!

Reply:
I'm going to reduce it to four to make it line with [ledger](https://github.com/alphagov/pay-ledger/pull/2204/files#diff-9adbcf1b657af3ada94e0c9cf58141d8191615da6c58d4d4ff6495f3afd70765R40).